packages/gatsby-remark-prismjs/src/highlight-code.js

```diff
@@ -36,10 +36,12 @@ module.exports = (language, code, lineNumbersHighlight = []) => {
     highlightedCode = ``
     // Don't add a new line character after highlighted lines as they
     // need to be display: block and full-width.
-    codeSplits.forEach(split => {
-      split.highlighted
-        ? (highlightedCode += split.code)
-        : (highlightedCode += `${split.code}\n`)
+    const maxCodeSplits = codeSplits.length - 1
+    codeSplits.forEach((split, idx) => {
+      highlightedCode += split.code
+      if (!split.highlighted && idx < maxCodeSplits) {
```

Comment:
I don't understand this change — if I'm reading this right, you changed it so that only the last split that's not highlighted has a new line character added?

Reply:
Isn't the issue just we're adding a code split to InlineCode blocks and we never should do that? Is there some way of detecting if we're highlighting an InlineCode block?